### DISCUSSION TITLE
Problem restoring Greenplum backups on CloudberryDB

### DISCUSSION THREAD

**Discussion by svnZA** [apache/cloudberry#737] [Q&A] [answered]
Good morning,

Firstly, I want to express my gratitude for the fantastic work you’re doing, it's truly appreciated.

We are in the process of transitioning from Greenplum 6.17.1 to CloudberryDB 1.6.0. My cluster is already set up and running (1 x Master, 1 x Master Standby, and 8 x Segment servers). However, I’m encountering the following error when attempting to restore my Greenplum backups on CloudberryDB:

**[CRITICAL]:-gprestore 1.6.0 cannot restore a backup taken with gpbackup 1.20.4; please use gprestore 1.20.4 or later.**

Your assistance in resolving this issue would be greatly appreciated.

Thank you in advance for your support.

Best regards,
Henk V

**Comment by my-ship-it** [comment]
Hi @svnZA, did you use gpbackup to backup Greenplum 6.17.1 provided by Cloudberry?

**Reply by svnZA** [reply]
Good morning,

Yes, that is what I used.

gpbackup version 1.20.4

gpbackup --dbname "cdrfw" --compression-level 9 --metadata-only --verbose --backup-dir /backup/gpbackups

**Reply by my-ship-it** [reply]
You can go to Slack for more help, where there are many enthusiastic people

**Reply by svnZA** [reply]
I noticed the following on the roadmap "Migration tool to help users migrate from Greenplum to Cloudberry Database."

But surely there has been a number of people moving from GP to CloudberryDB, so this can't be a new thing :)

**Comment by jianlirong** [comment, marked as answer]
Actually, there indeed is an open-sourced migration tool:

https://github.com/HashDataInc/cbcopy

You can have a try.

Thanks,
Lirong


svnZA ***@***.***> 于2024年11月28日周四 16:45写道：

**Reply by svnZA** [reply]
Thank you, much appreciated.

**Comment by xanadu-dev** [comment]
> Good morning,
> 
> Firstly, I want to express my gratitude for the fantastic work you’re doing, it's truly appreciated.
> 
> We are in the process of transitioning from Greenplum 6.17.1 to CloudberryDB 1.6.0. My cluster is already set up and running (1 x Master, 1 x Master Standby, and 8 x Segment servers). However, I’m encountering the following error when attempting to restore my Greenplum backups on CloudberryDB:
> 
> **[CRITICAL]:-gprestore 1.6.0 cannot restore a backup taken with gpbackup 1.20.4; please use gprestore 1.20.4 or later.**
> 
> Your assistance in resolving this issue would be greatly appreciated.
> 
> Thank you in advance for your support.
> 
> Best regards, Henk V

I faced a similar problem and simply disabled the version check, changing it from an error to a warning so that the process would continue - the restore worked perfectly fine.  See https://github.com/apache/cloudberry-gpbackup/pull/8

**Reply by my-ship-it** [reply]
> > Good morning,
> > Firstly, I want to express my gratitude for the fantastic work you’re doing, it's truly appreciated.
> > We are in the process of transitioning from Greenplum 6.17.1 to CloudberryDB 1.6.0. My cluster is already set up and running (1 x Master, 1 x Master Standby, and 8 x Segment servers). However, I’m encountering the following error when attempting to restore my Greenplum backups on CloudberryDB:
> > **[CRITICAL]:-gprestore 1.6.0 cannot restore a backup taken with gpbackup 1.20.4; please use gprestore 1.20.4 or later.**
> > Your assistance in resolving this issue would be greatly appreciated.
> > Thank you in advance for your support.
> > Best regards, Henk V
> 
> I faced a similar problem and simply disabled the version check, changing it from an error to a warning so that the process would continue - the restore worked perfectly fine. See [apache/cloudberry-gpbackup#8](https://github.com/apache/cloudberry-gpbackup/pull/8)

Thanks, we will have a look.

**Comment by my-ship-it** [comment]
Generally speaking, gpbackup is mainly used for database backup, not for upgrading, because different versions may have potential data compatibility issues.
If upgrading is needed, it is recommended to use the approach mentioned above.
